internal/service/domain/tree/tree.go

```diff
@@ -178,42 +192,34 @@ func (s *TreeService) Update(ctx context.Context, id int32, tu *entities.TreeUpd
 		return nil, handleError(err)
 	}
 
-	// Check if the tree is readonly (imported from csv)
-	// if currentTree.Readonly {
-	// 	return nil, handleError(fmt.Errorf("tree with ID %d is readonly and cannot be updated", id))
-	// }
-
-	fn := make([]entities.EntityFunc[entities.Tree], 0)
+	var treeCluster *entities.TreeCluster
 	if tu.TreeClusterID != nil {
-		var treeCluster *entities.TreeCluster
 		treeCluster, err = s.treeClusterRepo.GetByID(ctx, *tu.TreeClusterID)
 		if err != nil {
 			return nil, handleError(fmt.Errorf("failed to find TreeCluster with ID %d: %w", *tu.TreeClusterID, err))
 		}
-		fn = append(fn, tree.WithTreeCluster(treeCluster))
-	} else {
-		fn = append(fn, tree.WithTreeCluster(nil))
 	}
 
+	var sensor *entities.Sensor
 	if tu.SensorID != nil {
-		var sensor *entities.Sensor
 		sensor, err = s.sensorRepo.GetByID(ctx, *tu.SensorID)
 		if err != nil {
 			return nil, handleError(fmt.Errorf("failed to find Sensor with ID %v: %w", *tu.SensorID, err))
 		}
-		fn = append(fn, tree.WithSensor(sensor))
-	} else {
-		fn = append(fn, tree.WithSensor(nil))
 	}
 
-	fn = append(fn, tree.WithPlantingYear(tu.PlantingYear),
-		tree.WithSpecies(tu.Species),
-		tree.WithNumber(tu.Number),
-		tree.WithLatitude(tu.Latitude),
-		tree.WithLongitude(tu.Longitude),
-		tree.WithDescription(tu.Description))
+	updatedTree, err := s.treeRepo.Update(ctx, id, func(tree *entities.Tree) (bool, error) {
+		tree.PlantingYear = tu.PlantingYear
+		tree.Species = tu.Species
+		tree.Number = tu.Number
+		tree.Latitude = tu.Latitude
+		tree.Longitude = tu.Longitude
+		tree.Description = tu.Description
+		tree.TreeCluster = treeCluster
+		tree.Sensor = sensor
```

Comment:
Same here, you can run the pre-definitions within this function and assign them directly to the tree entity